keras/layers/core.py

```diff
@@ -418,9 +418,11 @@ def get_output(self, train=False):
         elif self.mode == 'cos':
             l1 = self.layers[0].get_output(train)
             l2 = self.layers[1].get_output(train)
-            output, _ = theano.scan(lambda v1, v2: T.dot(v1, v2) / T.sqrt(T.dot(v1, v1) * T.dot(v2, v2)),
-                                    sequences=[l1, l2],
-                                    outputs_info=None)
+            # output, _ = theano.scan(lambda v1, v2: T.batched_tensordot(v1, v2, 1) / T.sqrt(T.batched_tensordot(v1, v1, 1) * T.batched_tensordot(v2, v2, 1)),
+            #                         sequences=[l1, l2],
+            #                         outputs_info=None)
```

Comment:
I believe the code is logic reasonable but actually it causes some errors in compilation...